packages/protocol/contracts/common/FeeHandler.sol

```diff
@@ -0,0 +1,549 @@
+    TokenState storage tokenState = tokenStates[tokenAddress];
+    tokenState.handler = handlerAddress;
+  }
+
+  function removeToken(address tokenAddress) external onlyOwner {
+    _removeToken(tokenAddress);
+  }
+
+  function _removeToken(address tokenAddress) private {
+    _deactivateToken(tokenAddress);
+    TokenState storage tokenState = tokenStates[tokenAddress];
+    tokenState.handler = address(0);
+  }
+
+  function _sell(address tokenAddress) private onlyWhenNotFrozen nonReentrant {
+    IERC20 token = IERC20(tokenAddress);
+
+    TokenState storage tokenState = tokenStates[tokenAddress];
+    require(tokenState.handler != address(0), "Handler has to be set to sell token");
+    FixidityLib.Fraction memory balanceOfTokenToBurn = FixidityLib.newFixed(
+      token.balanceOf(address(this)).sub(tokenState.toDistribute)
+    );
+
+    uint256 balanceToBurn = (burnFraction.multiply(balanceOfTokenToBurn).fromFixed());
+
+    uint256 contractBalance = token.balanceOf(address(this));
+
+    tokenState.toDistribute += (contractBalance.sub(balanceToBurn));
+
+    // small numbers cause rounding errors and zero case should be skipped
+    if (balanceToBurn < MIN_BURN) {
+      return;
+    }
+
+    if (dailyBurnLimitHit(tokenAddress, balanceToBurn)) {
+      // in case the limit is hit, burn the max possible
+      // TODO move to state
+      balanceToBurn = tokenState.currentDaySellLimit;
+      emit DailyLimitHit(tokenAddress, balanceToBurn);
+    }
+
+    token.transfer(tokenState.handler, balanceToBurn);
+    IFeeHandlerSeller handler = IFeeHandlerSeller(tokenState.handler);
+
+    handler.sell(
+      tokenAddress,
+      registry.getAddressForOrDie(GOLD_TOKEN_REGISTRY_ID),
+      balanceToBurn,
+      FixidityLib.unwrap(tokenState.maxSlippage)
+    );
+
+    tokenState.pastBurn = tokenState.pastBurn.add(balanceToBurn);
+    updateLimits(tokenAddress, balanceToBurn);
+
+    emit SoldAndBurnedToken(tokenAddress, balanceToBurn);
+  }
+
+  /**
+    @dev Distributes the available tokens for the specified token address to the fee beneficiary.
+    @param tokenAddress The address of the token for which to distribute the available tokens.
+  */
+  function distribute(address tokenAddress) external {
+    return _distribute(tokenAddress);
+  }
+
+  function _distribute(address tokenAddress) private onlyWhenNotFrozen nonReentrant {
+    require(feeBeneficiary != address(0), "Can't distribute to the zero address");
+    IERC20 token = IERC20(tokenAddress);
+    uint256 tokenBalance = token.balanceOf(address(this));
+
+    TokenState storage tokenState = tokenStates[tokenAddress];
+
+    // safty check to avoid a revert due balance
+    uint256 balanceToDistribute = Math.min(tokenBalance, tokenState.toDistribute);
+
+    if (balanceToDistribute == 0) {
+      // don't distribute with zero balance
+      return;
+    }
+
+    token.transfer(feeBeneficiary, balanceToDistribute);
+    tokenState.toDistribute = 0;
```

Comment:
```suggestion
    tokenState.toDistribute = tokenState.toDistribute.sub(balanceToDistribute);
```

Since we consider this case also on line 359 we should take it into account also here.

Reply:
Nice catch, in theory this shouldn't happen but it's true we should have it for consistency